### DISCUSSION TITLE
Taskchampion vs recurrence

### DISCUSSION THREAD

**Discussion by gour** [GothenburgBitFactory/taskwarrior#2831] [General]
Hello!

Yesterday when asking about 3.0 release, mostly related to  fixing (once and for all) current lack of proper support for recurring tasks, I was told that, probably, Taskchampion-related stuff have higher priority than releasing 3.0 with "recurrence overhaul". Few times I tried to move to Emacs/org-mode, but always returned back seeing TW as simply (much) more productive for me, but I was always bitten with the extra admin overhead in managing many recurring tasks.

For some time I was even sponsoring development hoping we'll see some beta by the end of the 2021, but then gave up, moved to [Etm](https://github.com/dagraham/etm-dgraham) which handles are recurring needs, but...TW's interface is still more productive, so I wonder how does the work on Taskchampion which seems to influence taskserver as well influence work on planned "recurrence overhaul" scheduled for 3.0?

How far/close is 3.0 release anyway?

Sincerely,
Gour

**Comment by angelus2014** [comment]
I would have to agree, the recurrence overhaul would need to be the highest priority as this, imho, is the biggest problem in TW.


---
With regards,

Angelo Machils

**Comment by djmitche** [comment]
I can only speak to the work on the sync overhaul, since I'm the person doing most of it.  It's quite far from done, so I can't put a date on it -- and certainly, if other work needs to ship before that is ready, that's OK too.

**Reply by gour** [reply]
> I can only speak to the work on the sync overhaul, since I'm the person doing most of it. It's quite far from done, so I can't put a date on it 

Thanks a lot for your work on reviving Taskwarrior!

> and certainly, if other work needs to ship before that is ready, that's OK too.

Well, my point is that `recurrence` is an important feature even for single-device users not having need to sync across many devices. :wink:

**Comment by tbabej** [comment]
I am taking the lead on the recurrence features. It's a big change, so putting timeline with limited resources is difficult, but I have local prototypes of the subset of the changes implemented for some time. To be clear, the sync overhaul, given @djmitche is taking lead on it, is not decreasing the probability of recurrence landing in 2022.

**Reply by gour** [reply]
> I am taking the lead on the recurrence features. It's a big change, so putting timeline with limited resources is difficult, but I have local prototypes of the subset of the changes implemented for some time. 

Great!

I've started to  use Taskwarrior again...

> To be clear, the sync overhaul, given @djmitche is taking lead on it, is not decreasing the probability of recurrence landing in 2022.

:+1:

**Reply by gour** [reply]
Just curious if, ten months (years :-) after, something is more clear in regard to recurrence support?

**Comment by gour** [comment]
> I can only speak to the work on the sync overhaul, since I'm the person doing most of it. It's quite far from done, so I can't put a date on it -- and certainly, if other work needs to ship before that is ready, that's OK too.

I've settled on openSUSE Tumbleweed for which there is an old `taskd` package, using taskdctl script, but no systemd service. Is there any use trying to update the package, or there is simply not future for `taskd´?

**Reply by djmitche** [reply]
I don't think there's a future for taskd, but there might be a present -- if it's just a matter of adding a systemd script to the package and bumping the package version, that seems a reasonable thing to do.
